publisher/internal/analytics/ethereum/operations.go

```diff
@@ -178,16 +171,19 @@ func (add *Addition) Extract(mint EventLog) error {
 	}
 	add.Position = *addPosition
 
-	for _, evLog := range txEventsFromCache.([]EventLog) { // Go through all events of this transaction
-		if isTransferEvent(evLog) && !isUniswapPositionsNFT(evLog.Address) {
-			add.handleLiquidityTransfer(mint, evLog)
-		}
+	transferLogs, err := add.cache.GetByTxHashAndLogType(mint.TransactionHash, eventInstructions[transferSig].Name)
+	if err != nil {
+		return err
+	}
+	for _, transferLog := range transferLogs { // Go through all transfers of this transaction
+		add.handleLiquidityTransfer(mint, transferLog)
 	}
 
-	if isEitherTokenUnknown(*addPosition) {
+	if isEitherTokenUnknown(add.Position) {
```

Comment:
Can `isEitherTokenUnknown` be a `Position` struct method? And named `areTokensSet`?

Reply:
Changed, should we move other methods from checks.go (isOrderCorrect, isStableOrNativeInvolved, isEitherTokenAmountZero) to Position struct methods as well?

Reply:
If they rely only on data contained in `Position` struct itself, then I don't see why not.